src/rules/require-source-directory.ts

```diff
@@ -0,0 +1,19 @@
+import { buildRule } from './helpers';
+import { RuleName } from './types';
+
+export default buildRule({
+  name: RuleName.RequireSourceDirectory,
+  description: 'Does the `src/` directory exist?',
+  dependencies: [],
+  execute: async ({ project, pass, fail }) => {
+    const entryPath = 'src';
+    const stats = await project.fs.getEntryStats(entryPath);
+    const passed = stats?.isDirectory();
+    const message = stats
+      ? `\`${entryPath}\` is not a directory when it should be.`
+      : `\`${entryPath}\` exists in the module template, but not in this repo.`;
```

Comment:
Nit: Maybe we could simplify this message a bit

```suggestion
      : `\`${entryPath}\` does not exist in this repo.`;
```

The rule doesn't actually check the module template after all, so it might be confusing if we ever restructure the template

Reply:
Updated here: [bc3f58e](https://github.com/MetaMask/module-lint/pull/33/commits/bc3f58e12c001be24380104f9e83fd78f38d1780)